datasets/utils/secret.py

```diff
@@ -0,0 +1,210 @@
+import base64
+import json
+import logging
+import os
+from collections import defaultdict
+from dataclasses import dataclass
+from typing import Dict, Union
+
+import boto3
+from kubernetes import client, config
+from tenacity import (
+    Retrying,
+    retry_if_exception,
+    stop_after_attempt,
+    wait_fixed,
+)
+
+
+logger = logging.getLogger(__name__)
+
+# Ensure we return a non-None value
+# Prefer to throw exception rather than returning None if secret can't be found
+secret_return_type = Union[str, Dict[str, str]]
+
+# Global secret cache to prevent redundant
+# secret retrieval calls to external system if possible
+secret_cache = defaultdict(dict)
+
+
+@dataclass
+class Secret:
+    """
+    Dataclass interface for secret retrieval.
```

Comment:
This dataclass is actually for secret retrieval only.  So naming it `Secret` could mean more.

Maybe `SecretFetcher` ? That would reduce confusion that it could also store secrets.

Thoughts?

Reply:
Yeah I was thinking more on the overall secret lifecycle where there are also topics like secret generation and register, but agreed for now we can rename to `SecretFetcher` and reevaluate in the future

Reply:
how about rename the file to secret_fetcher?